src/components/Event/EventSearchBar.tsx

```diff
@@ -57,81 +56,45 @@ const EventSearchBar = ({
     return uniqueResults
   }
 
-  useEffect(() => {
-    const query = router.query
-    if (query.title || query.startDate || query.endDate) {
-      setSearchActive(true)
-      const arg = {
-        title: (query.title as string) || '',
-        startDate: (query.startDate as string) || undefined,
-        endDate: (query.endDate as string) || undefined,
+  const debouncedFetchEventSearch = useCallback(
+    debounce(async (arg: TQuerySearch) => {
+      try {
+        const results = await fetchEventSearch(arg)
+        setEventData(results)
+      } catch (err) {
+        console.error(err)
+      } finally {
+        setIsLoading(false)
       }
+    }, 500),
+    [fetchEventSearch, setEventData, setIsLoading],
+  )
 
-      const activeModifiers: ActiveModifiers = {} // assuming this is the correct type
-      const mouseEvent = {} as React.MouseEvent<Element, MouseEvent> //
-      if (query.title) setSearchQuery(query.title)
-      if (query.startDate) {
-        const startDate = new Date(query.startDate as string)
-        setSearchDate(
-          { from: startDate, to: undefined },
-          startDate,
-          activeModifiers,
-          mouseEvent,
-        )
-      }
-      if (query.endDate) {
-        const startDate = new Date(query.startDate as string)
-        const endDate = new Date(query.endDate as string)
-        setSearchDate(
-          { from: startDate, to: endDate },
-          startDate,
-          activeModifiers,
-          mouseEvent,
-        )
-      }
+  useEffect(() => {
+    const shouldSearch =
+      searchQuery.length >= 3 || searchDate?.from || searchDate?.to
 
+    if (shouldSearch) {
       setIsLoading(true)
-      fetchEventSearch({ arg })
-        .then((response) => {
-          setEventData(response)
-        })
-        .catch((err) => console.error(err))
-        .finally(() => setIsLoading(false))
-    }
-  }, [router.query])
-
-  const submitHandler = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    setSearchActive(true)
-
-    const queryParams: {
-      title?: string
-      startDate?: string
-      endDate?: string
-    } = {}
-
-    if (searchQuery) queryParams.title = searchQuery
-    if (searchDate?.from) queryParams.startDate = dateFormater(searchDate.from)
-    if (searchDate?.to) queryParams.endDate = dateFormater(searchDate.to)
+      setSearchActive(true)
 
-    router.push(
-      {
-        pathname: router.pathname,
-        query: queryParams,
-      },
-      undefined,
-      { shallow: true },
-    )
-  }
+      debouncedFetchEventSearch({
+        title: searchQuery,
+        startDate: searchDate?.from ? dateFormater(searchDate.from) : undefined,
+        endDate: searchDate?.to ? dateFormater(searchDate.to) : undefined,
```

Comment:
Since `searchDate` is optional, you want to use conditional chaining always or it will throw an error some times
```suggestion
        startDate: searchDate?.from ? dateFormater(searchDate?.from) : undefined,
        endDate: searchDate?.to ? dateFormater(searchDate?.to) : undefined,
```